### DISCUSSION TITLE
Feature Request: File Upload Capability for GitHub Action Inputs

### DISCUSSION THREAD

**Discussion by herzallah** [community/community#121198] [Actions]
### Select Topic Area

Product Feedback

### Body

**Overview**:
This proposal advocates for the introduction of a file upload input within the workflow_dispatch trigger for GitHub Actions, enhancing the tool's capability to handle workflows that require external files directly.

**Feature Details:**
- **Input Name:** upload_file
- **Description**: Enables users to securely upload necessary files for workflow execution.
- **Attributes:**
  - **Type:** file
  - **Default:** 'Specify a default file path or leave blank for user input'

**Use Cases:**
Users can upload scripts, configuration files, secret file, or other necessary documents that are crucial for their automated processes.

**Benefits**:
This feature will allow for more dynamic and versatile workflows, accommodating various user needs by facilitating the direct upload of essential files.

**Implementation Suggestions:**
The functionality should be incorporated into the GitHub Actions UI, ensuring that file uploads are handled securely and efficiently, with files being available only where needed in the workflow.

**Conclusion**:
By enabling file uploads directly in GitHub Actions, the platform will significantly enhance its utility for complex automations, providing users with the flexibility to easily incorporate necessary external files into their workflows.

**Comment by grandmas-favorite** [comment]
Hi @herzallah!

Thanks for sharing and we appreciate your patience. You can check our [public roadmap](https://github.com/github/roadmap/projects/1) to see the kinds of work we are focusing on for the coming year. Smaller fixes won’t be listed, but are often included when work is happening adjacent to it.

As a GitHub user, the most you can do is to share your feedback on this Discussion space with the correct tags (include examples), and trust that we are always working to make GitHub as awesome as possible. We can't guarantee that every bit of feedback sees action, but we are paying attention!
